businessRules/donations.js

```diff
@@ -1,54 +1,59 @@
 const { v4 } = require('uuid');
 const { createDonation } = require('../models/donations');
-const { getAllIdeas } = require('../models/ideas');
-const { getIdeas } = require('../utils/storage');
+const { getIdea, validateIdea } = require('../validations/ideas');
+const { validateSum } = require('../validations/donations.js');
+const { getUser } = require('../validations/users');
 
-function createDonation(firstName, sum, userId, ideaId) {
-    let ideas = getIdeas();
-    const idea = ideas.find((idea) => ideaId === idea.id);
+function createDonationByUnregisteredUser(sum, userId, ideaId) {
+    validateIdea(ideaId);
+    validateSum(sum);
 
-    if (!idea) {
-        throw new Error('Error. No idea with such ID found.');
+    const user = getUser(userId);
+
+    let firstName;
+
+    if (!user) {
+        const generateId = v4();
+        const anonymous = 'Anonymous';
+
+        firstName = firstName || anonymous;
+        userId = userId || generateId;
```

Comment:
You can wrap these 4 lines into a new function: `createAnonymousUser` and reuse it here

Reply:
Now here, I'm struggling with function. Have some troubles.